### ISSUE TITLE
Seq Lab: New Python class for Smartsheet QC data

### ISSUE THREAD

**Issue by tim-snyder-wustl** [GTAC-MGI/GTAC-ESP-LIMS#238]
Blocked by:
https://github.com/GTAC-MGI/GTAC-ESP-LIMS/issues/236

Relates to:
https://github.com/GTAC-MGI/GTAC-ESP-LIMS/issues/239

New class for updating Smartsheet (build a new sheet in Smartsheet, in addition: QC complete action (per sample), Move to DT action (per sample group / SR#))

Add data for DT - filepaths, data receiver

Check if SR exists, if not, create Parent row with high level project data including DT/Analaysis reqs.
Then add children rows of samples that have completed QC w/QC metrics.

If SR exists, iterate over child sample rows, if child sample already in SS update QC data, if not, add new child row with QC data.

Add pass/fail status looking at QC metrics.

---

We need from Mark:

Filepaths for FASTQ files, and which sample they belong to
The actual QC stats
SR# for samples
Show these Library QC Metrics:

https://wustl.app.box.com/file/950842021254

Example:

https://app.smartsheet.com/sheets/J7g3JgmFw7QFQmcFHqxqJFPF9p38rc7mg6GgJ541?view=grid

There should be combined totals for libraries that may have been run over multiple lanes as well.

Need

- SR # associated with QC metrics
- Sample Name / ID

**Comment by billtell** [comment]
Framework created. Basic testing completed.  
Need to add more specific fields once example json is provided.  Will create a new ticket to add specifics.
